includes/Admin/PluginActionLinks.php

```diff
@@ -63,13 +63,14 @@ public static function get_registration_action(): string {
 	 *
 	 * @since 1.6.0
 	 *
-	 * @param  array $links Plugin action links.
+	 * @param  array|mixed $links Plugin action links.
 	 *
-	 * @return array
+	 * @return array|mixed
 	 */
-	public function action_links( $links ): array {
+	public function action_links( $links ) {
 		$slug    = sprintf( 'edit.php?post_type=%s&page=stories-dashboard#/editor-settings', Story_Post_Type::POST_TYPE_SLUG );
 		$url     = get_admin_url( null, $slug );
+		$links   = (array) $links;
```

Comment:
Since `$links` is cast to an array, the return type will always be an array, so we could actually now retain the return type.

Reply:
Same probably applies to more places.

Reply:
This is true of a lot of places. But I think it is cleaner to have a hard rule of not having a type hint on any filter callback. If we start doing this in same places and not others, it would get confusing. Thoughts?

Reply:
I'd like our type declarations to be as specific as possible.

When we are certain about the return type, it's clear to specify this with a return type.

The alternative would be to not do type casting at all because of the drawbacks as per https://github.com/google/web-stories-wp/pull/8724/files#r692021032.

So instead of `$links = (array) $links;` we would do `if ( ! is_array( $links ) ) { return $links; }`

Then we can't add a return type obviously.

Reply:
> So instead of $links = (array) $links; we would do if ( ! is_array( $links ) ) { return $links; }

So instead of casting, should I replace that with type checking?

Reply:
Well that depends on whether it makes sense or not. I would like to hear your opinion on it so we can outweigh pros and cons of the three different approaches (no return types + casting), return types + casting, no return types + no casting).

Reply:
return types + casting
In my cases, there are mixed return types, as casting is not possible. We could add it in some places or cast the return. But there we will have to make a note of where this casting should and should not be. 

no return types + casting
No return types means we can have mixed returned values, which most of these filters are likely to be. Casting ensure, that we do, say a array_merge, that we don't get e fatal error. 

no return types + no casting
This can result in fatal errors, as noted above. This is the one that most likely to generate problems.

Reply:
> But there we will have to make a note of where this casting should and should not be.

I think the code itself and the PHPDoc documentation will make this quite clear. Plus PHPStan and PHPCS would yell at us otherwise.

> no return types + no casting

To clarify, with this I meant not doing casting, _but_ doing type checks like `is_array()` instead. To avoid fatal errors.

Reply:
> but doing type checks like is_array() instead.

If we do this, our filters will do nothing...

Reply:
Doing nothing if another plugin does bad things is better than crashing or causing notices due to unexpected behavior.

Reply:
After letting this sink in for a bit, I think we should proceed with the `if ( ! is_array( $links ) ) { return $links; }` approach.

If a plugin filters `$links` to be something else, then something is really broken and we should not bother trying to cast to an array. Especially because of the unintended side effects casting could have (as per https://github.com/google/web-stories-wp/pull/8724/files#r692021032)